packages/flutter_driver/lib/src/common/wait.dart

```diff
@@ -0,0 +1,185 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:flutter/scheduler.dart';
+
+/// Base class for a class that can be serialized to JSON.
+abstract class Jsonable {
+  /// Serializes the object to JSON.
+  Map<String, dynamic> serialize();
+}
+
+/// Thrown to indicate a JSON serialization error.
+class SerializationException implements Exception {
+  /// Creates a [SerializationException] with an optional error message.
+  SerializationException([this.message]);
+
+  /// The error message, possibly null.
+  final String message;
+
+  @override
+  String toString() => 'SerializationException($message)';
+}
+
+/// Base class for a condition that can be waited upon.
+abstract class WaitCondition extends Jsonable {
+  /// Retrieves the condition that is waited upon.
+  ///
+  /// The future returned by [wait] should be finished if this condition turns
+  /// true.
+  bool get condition;
+
+  /// Returns a future that will be completed until [condition] turns true.
+  Future<void> wait();
+}
+
+/// A condition that waits until no transient callbacks scheduled.
+class NoTransientCallbacksCondition implements WaitCondition {
+  /// Creates a [NoTransientCallbacksCondition] instance.
+  const NoTransientCallbacksCondition();
+
+  /// Factory constructor to parse a [NoTransientCallbacksCondition] instance
+  /// from the given JSON map.
+  factory NoTransientCallbacksCondition.fromJson(Map<String, dynamic> json) {
```

Comment:
Aren't these called `deserialize` in the context of flutter driver commands? Maybe use same terminology?